File: atom-sdk/AtomSDK/AtomSDKTests/BatchTests.m
//
//  BatchTests.m
//  AtomSDK
//
//  Created by Valentine.Pavchuk on 12/12/16.
//  Copyright © 2016 IronSource. All rights reserved.
//

#import <XCTest/XCTest.h>

@import AtomSDK;

@interface BatchTests : XCTestCase

@end

@implementation BatchTests

- (void)setUp {
    [super setUp];
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testInit {
    // This is an example of a functional test case.
    // Use XCTAssert and related functions to verify your tests produce the correct results.
    
    NSArray* expectedEvents = [NSArray arrayWithObjects: @"test 1", @"test 2", nil];
    int expectedId = 42;
    
    let batchObject = IS(events: expectedEvents, lastId: expectedId)
    
    XCTAssertEqual(expectedId, batchObject.lastId)
    XCTAssertEqual(expectedEvents.count, batchObject.events.count)
    
    for i in 0...(expectedEvents.count - 1) {
        XCTAssertEqual(expectedEvents[i], batchObject.events[i])
    }
}

@end

File: atom-sdk/AtomSDK/AtomSDKTests/BatchTests.swift
//
//  BatchTests.swift
//  AtomSDK
//
//  Created by g8y3e on 6/22/16.
//  Copyright © 2016 IronSource. All rights reserved.
//

import XCTest
@testable import AtomSDK

class BatchTests: XCTestCase {

    override func setUp() {
        super.setUp()
    }
    
    override func tearDown() {
        super.tearDown()
    }

    func testInit() {
        let expectedEvents = ["test 1", "test 2"]
        let expectedId: Int32 = 42
        
        let batchObject = Batch(events: expectedEvents, lastId: expectedId)
        
        XCTAssertEqual(expectedId, batchObject.lastId)
        XCTAssertEqual(expectedEvents.count, batchObject.events.count)
        
        for i in 0...(expectedEvents.count - 1) {
            XCTAssertEqual(expectedEvents[i], batchObject.events[i])
        }
    }
}
